### PR TITLE
Add main contract price functions

### LINKED ISSUE #40: Add main contract price functions
Integrate into the main contract. This will help with the integrations into other systems.

```
@public
@constant
def tokenOutputAmountAfterFees(input_token_amount: uint256, input_token_address: address, output_token_address: address) -> uint256:
```

and

```
@public
@constant
def tokenExchangeRateAfterFees(input_token_address: address, output_token_address: address) -> uint256:
```

`tokenOutputAmountAfterFees` should use `tokenExchangeRateAfterFees`.

Should be covered by tests.

`swapTokens` should also be simplified after these changes.

### PR DESCRIPTION
**Pull request by esaulkov**
resolves #40

### CODE REVIEW

File: contracts/stablecoinswap.vy [hunk 1]
```diff
@@ -128,6 +128,29 @@ def removeLiquidity(token_address: address, stableswap_token_amount: uint256, er
 
     return True
 
+@public
+@constant
+def tokenExchangeRateAfterFees(input_token_address: address, output_token_address: address) -> uint256:
+    assert self.inputTokens[input_token_address] and self.outputTokens[output_token_address]
```
Comment (gonchs) [inline contracts/stablecoinswap.vy]: We already have the assert in swapTokens. I think we can remove it from here and from tokenOutputAmountAfterFees to save gas.

**Comment by gonchs** [comment]
@blimp666 Can you check and also compare the gas cost of the swap relative to the previously deployed version?

**Comment by blimp666** [comment]
@esaulkov do we already have a new contract deployed to Kovan?

**Comment by esaulkov** [comment]
No

**Comment by blimp666** [comment]
Ok, I'll deploy it

**Comment by gonchs** [comment]
@esaulkov Can we adjust the code to fit the following?

```
@public
@constant
def tokenOutputAmountAfterFees(input_token_amount: uint256, input_token_address: address, output_token_address: address) -> uint256:
    output_token_amount: uint256 = input_token_amount * self.tokenExchangeRateAfterFees(input_token_address, output_token_address)
    return output_token_amount
```

`tokenExchangeRateAfterFees` should return the final price multiplier so it should be possible to just do `output_token_amount = input_token_amount * tokenExchangeRateAfterFees(...)`

This will make it easier to use for 3rd party developers since it doesn't require any further decimal multiplications.

**Comment by esaulkov** [comment]
Let's assume both tokens have the same price 1 USD and our fees are equal to 0. Then according your scenario we have:
```
1 DAI * tokenExchangeRateAfterFees() = 1 USDC
1 * 10**18 * tokenExchangeRateAfterFees() = 1 * 10**6
```
and `tokenExchangeRateAfterFees()` should return `10**(-12)`. I tried to use `Decimal` type in `tokenExchangeRateAfterFees()` but it returned `0` instead value above. The cause is this type has precision of 10 decimal places only.

Also I tried to calculate `tokenExchangeRateAfterFees` in `ether` instead `wei`. In this case for example from test (1 DAI = 1.01 USD, 1 USDC = 0.99 USD) we have:
```
1 DAI = 1.017141 USDC
1 USDC = 0.977257425742574257 DAI
```
But `Decimal` type returns `0.9772574257` due the same issue - this type has 10 decimal places only.

**Comment by gonchs** [comment]
OK, let's keep the current implementation. We'll get back to this after https://github.com/stablecoinswap/stablecoinswap-contracts/issues/44

**Pull request merged**